### ISSUE TITLE
azurerm_redis_cache > Basic fails

### ISSUE THREAD

**Issue by dhabierre** [hashicorp/terraform-provider-azurerm#18877]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.14.5

### AzureRM Provider Version

2.57.0

### Affected Resource(s)/Data Source(s)

azurerm_redis_cache

### Terraform Configuration Files

```hcl
resource "azurerm_redis_cache" "redis_cache" {
  name                          = "..."
  location                      = "..."
  resource_group_name           = "..."
  capacity                      = 0
  family                        = "C"
  sku_name                      = "Basic"
  public_network_access_enabled = false
  private_static_ip_address     = "10.25..."
  enable_non_ssl_port           = false
  minimum_tls_version           = "1.2"
}
```

### Debug Output/Panic Output

```shell
15:00:42  azurerm_redis_cache.redis_cache[0]: Creating...
15:00:43  
15:00:43  Error: creating Redis Cache "[...]" (Resource Group "[...]-rec01"): redis.Client#Create: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidRequestBody" Message="The value of the parameter 'properties.subnetId' is invalid.\r\nRequestID=[...]"
```

### Expected Behaviour

The resource should be created.

### Actual Behaviour

```
The value of the parameter 'properties.subnetId' is invalid.
```

**Comment by dhabierre** [comment]
@tombuildsstuff, I don't understand your answer at [11430](https://github.com/hashicorp/terraform-provider-azurerm/issues/11430). Is there any solution? Thanks!

**Comment by xuzhang3** [comment]
@dhabierre  to disable the `public_network_access_enabled` or set `private_static_ip_address` you need specific a `azurerm_virtual_network` and `subnet` for the redis cache
```hcl
resource "azurerm_resource_group" "example" {
	name     = "test"
	location = "eastus"
  }
  
  resource "azurerm_virtual_network" "example" {
	name                = "test"
	address_space       = ["10.0.0.0/16"]
	location            = azurerm_resource_group.example.location
	resource_group_name = azurerm_resource_group.example.name
  }
  
  resource "azurerm_subnet" "example" {
	name                 = "test"
	resource_group_name  = azurerm_resource_group.example.name
	virtual_network_name = azurerm_virtual_network.example.name
	address_prefixes     = ["10.0.2.0/24"]
  }
  
  
  resource "azurerm_redis_cache" "redis_cache" {
	name                          = "test"
	location                      = azurerm_resource_group.example.location
	resource_group_name           = azurerm_resource_group.example.name
	capacity                      = 1
	family                        = "P"
	sku_name                      = "Premium"
	public_network_access_enabled = false
	private_static_ip_address     = "10.0.2.19"
	enable_non_ssl_port           = false
	minimum_tls_version           = "1.2"
  
	subnet_id = azurerm_subnet.example.id
  }
  
```

**Comment by dhabierre** [comment]
Hi @xuzhang3 and thanks for your reply.

Ok, `subnet_id` value is required when using `private_static_ip_address`, right!

[azurerm_redis_cache](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/redis_cache#subnet_id) documentation says:

```
subnet_id - (Optional) Only available when using the Premium SKU The ID of the Subnet within which the Redis Cache should be deployed. This Subnet must only contain Azure Cache for Redis instances without any other type of resources.
```

Strange thing by the way: `This Subnet must only contain Azure Cache for Redis instances without any other type of resources.`

In my subnet there are VMs.
So according to this, I could'nt use Basic sku with static private IP?
Right?

Thanks.

**Comment by xuzhang3** [comment]
@dhabierre yes, according to this doc you need upgrade the SKU to `Premium` if you need `private_static_ip_address` and`subnet_id`. Also subnet can only contains  Redis caches. I tried with create a VM with the same subnet, the resource can be created but I think there's some issues with the network/connection not tried that.

**Comment by dhabierre** [comment]
Hi @xuzhang3,

Thanks for these details 👍
